Source/Modules/r.cxx

```diff
@@ -2141,7 +2145,18 @@ int R::functionWrapper(Node *n) {
   if (destructor)
     Printv(f->code, "R_ClearExternalPtr(self);\n", NIL);
 
-  Printv(f->code, "return r_ans;\n}\n", NIL);
+  Printv(f->code, "return r_ans;\n", NIL);
+  
+  /* Error handling code */
+  Printv(f->code, "fail: SWIGUNUSED;\n", NIL);
+  if (need_cleanup) {
+    Printv(f->code, cleanup, NIL);
+  }
+  Printv(f->code, "  Rf_error(\"%s %s\", SWIG_ErrorType(SWIG_lasterror_code), SWIG_lasterror_msg);\n", NIL);
```

Comment:
Previously:

```
#define SWIG_Error(code, msg) Rf_warning(msg); return Rf_ScalarLogical(NA_LOGICAL)
```

Now it's `Rf_error()` not `Rf_warning()`.  I know nothing useful about R, but that seems like a change that deserves explanation...

Also, `SWIG_Error()` used to return from the function, but now execution continues to whatever follows.  I wonder if `SWIG_Error()` ought to be a macro which calls a helper function (to do what the new `SWIG_Error()` function does) and then does `SWIG_fail;`?